File: site/DFM.Generic/Crypt.cs
using System;
using BCrypt.Net;
using crypt = BCrypt.Net.BCrypt;

namespace DFM.Generic
{
	public static class Crypt
	{
		public static String Do(String text)
		{
			return crypt.HashPassword(text);
		}

		public static Boolean Check(String text, String hash)
		{
			try
			{
				return crypt.Verify(text, hash);
			}
			catch (SaltParseException)
			{
				return false;
			}
		}
	}
}

File: site/Tests/Browser/helpers/crypt.js
const bcrypt = require('bcrypt');

const saltRounds = 10;

const hash = async function(text) {
	return await bcrypt
		.hash(text, saltRounds)
		.then(hash => hash)
}

const check = async function(text, hash) {
	return await bcrypt
		.compare(text, hash)
		.then(isRight => isRight)
}

module.exports = { hash, check }
